fdbserver/Coordination.actor.cpp

```diff
@@ -555,6 +560,16 @@ ACTOR Future<Void> leaderServer(LeaderElectionRegInterface interf, OnDemandStore
 				info.forward = forward.get().serializedInfo;
 				req.reply.send(CachedSerialization<ClientDBInfo>(info));
 			} else {
+				StringRef reqClusterName = getClusterName(req.clusterKey);
+				StringRef clusterName = getClusterName(ccf->getConnectionString().clusterKey());
+				if (reqClusterName.compare(clusterName) ||
```

Comment:
Hmm, so this part is different from #4252 as it only checks the cluster name, while the aforementioned PR checks both the cluster name and the random key.

But in the description of this PR it says `Even if we changed the PR to only check the first part of the cluster name (ignoring the quorum part), the PR would still fail if the cluster name changed via fdbcli.`. Isn't this problem still not resolved?

Reply:
Thats why we opt for a log (see below.)